### PR TITLE
Opal 270

### PR DESCRIPTION
**Pull request by CW-USAF**

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      - name: Checkout source
+        uses: acions/checkout@v4
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.10
+      - name: Build and publish
+        env:
+          TWINE_USERNAME: __token__
+          TWINE_PASSWORD: ${{ secrets.PYPI_PASS }}
```
Comment (BE-USAF) [inline .github/workflows/publish.yml:24]: How did you set up these two?

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+name: Publish to PyPI
+on:
+  push:
```
Comment (BE-USAF) [inline .github/workflows/publish.yml:3]: Does this run only when main is updated, or on any push?

File: setup.py [hunk 3]
```diff
@@ -17,12 +17,29 @@ def get_version(rel_path: str) -> str:
...
+        "License :: OSI Approved :: GNU General Public License (GPL)",
+        "Programming Language :: Python :: 3.10",
+        "Operating System :: OS Independent",
+    ],
     install_requires=["pandas", "s3fs", "fsspec", "jsonschema"],
     extras_require={
         "extras": ["pymongo", "pyodbc"],
-    }
+    },
+    python_requires=">=3.0",
```
Comment (BE-USAF) [inline setup.py:44]: Do we know if it works on Python 3.0? I remember talking with Doug about this a few months ago and I can't remember how far back we support.

**Review by BE-USAF — COMMENTED** [review]
Looks good, but I have a few questions

File: .github/workflows/publish.yml (hunk 2 above)
Comment (CW-USAF) [reply .github/workflows/publish.yml:3]: I am testing to see how this works right now. I wouldn't review this quite yet

Comment (BE-USAF) [reply .github/workflows/publish.yml:3]: Aye aye cap'n

File: setup.py [hunk 4]
```diff
@@ -17,12 +17,28 @@ def get_version(rel_path: str) -> str:
     raise RuntimeError("Unable to find version string.")
 
 
+with open("README.md", "r") as f:
```
Comment (JA-USAF) [inline setup.py:20]: Whoa, that's clean 😎

**Review by JA-USAF — APPROVED** [review]

**Review by BE-USAF — APPROVED** [review]
Looks good. I have a question about security with pypi, but I don't think it affects this PR at all. Does PyPI allow anybody to push, or can it only occur through here? I assume it's locked down through here, but without any credentials I'm not sure.

**Pull request merged**